fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/internal/JAXRSArchiveImpl.java

```diff
@@ -99,8 +112,62 @@ private static boolean hasApplicationPathAnnotation(Archive<?> archive) {
         return false;
     }
 
+    private static boolean hasApplicationServletMapping(Archive<?> archive) {
+        Node webXmlNode = archive.get(PATH_WEB_XML);
+        if (webXmlNode != null) {
+            return hasApplicationServletMapping(webXmlNode.getAsset());
+        }
+        return false;
+    }
+
+    private static boolean hasApplicationServletMapping(Asset webXmlAsset) {
```

Comment:
> Instead of this XML parsing and XPath,...

Well, I think that parsing XML is more robust - in theory, you don't have to use `WebXmlAsset` but e.g. a `FileAsset`.

> Or add more accessors if we you need...

I don't understand. Could you explain in more details?